src/utils/mod.rs

```diff
@@ -83,6 +84,85 @@ pub fn parse_invite(code: &str) -> &str {
     }
 }
 
+/// Retreives Id from a username mention.
+pub fn parse_username(mention: &str) -> Option<u64> {
+    if mention.len() < 4 {
+        return None;
+    }
+    if mention.starts_with("<@!") {
+        let len = mention.len() - 1;
+        mention[3..len].parse::<u64>().ok()
+    } else if mention.starts_with("<@") {
+        let len = mention.len() - 1;
+        mention[2..len].parse::<u64>().ok()
+    } else {
+        None
+    }
+}
+
+/// Retreives Id from a role mention.
+pub fn parse_role(mention: &str) -> Option<u64> {
+    if mention.len() < 4 {
+        return None;
+    }
+    if mention.starts_with("<@&") {
+        let len = mention.len() - 1;
+        mention[3..len].parse::<u64>().ok()
+    } else {
+        None
+    }
+}
+
+/// Retreives Id from a channel mention.
+pub fn parse_channel(mention: &str) -> Option<u64> {
+    if mention.len() < 4 {
+        return None;
+    }
+    if mention.starts_with("<@#") {
+        let len = mention.len() - 1;
+        mention[3..len].parse::<u64>().ok()
+    } else {
+        None
+    }
+}
+
+/// Retreives name and Id from an emoji mention.
+pub fn parse_emoji(mention: &str) -> Option<EmojiIdentifier> {
+    let len = mention.len();
+    if len < 4 || len > 100 {
```

Comment:
This will incorrectly return None on emojis which have a name of around 80+ chars (with a max of 100), won't it? Since the Id is at least 17 digits, and `<: : >` comprises 4 chars?